packages/flutter/lib/src/material/material.dart

```diff
@@ -593,7 +607,10 @@ class _MaterialInteriorState extends AnimatedWidgetBaseState<_MaterialInterior>
   @override
   Widget build(BuildContext context) {
     return new PhysicalShape(
-      child: widget.child,
+      child: new CustomPaint(
+        child: widget.child,
+        foregroundPainter: new _ShapeBorderPainter(_border.evaluate(animation)),
```

Comment:
we should cache the call to _border.evaluate rather than calling it twice

Reply:
done

Reply:
done